### ISSUE TITLE
All handlers succeeded for creation despite raising kopf.HandlerFatalError

### ISSUE THREAD

**Issue by moltzaum** [zalando-incubator/kopf#387]
## Long story short

Despite raising `kopf.HandlerFatalError`, I get 'All handlers succeeded for creation'. I want the handler to fail and the custom resource to get destroyed (so that there is no conflict when I try again).

## Description

I am hoping that I'm just misunderstanding something and can change what I'm doing.

I have used both `kopf.HandlerFatalError` as well as `kopf.PermanentError`. I have run from my machine as well as a kubernetes deployment. Its something I've ignored for the most part so far, but it seems like a bug.

<details><summary>The code snippet to reproduce the issue</summary>

```python
raise kopf.HandlerFatalError("Missing required ConfigMap")
```
</details>

<details><summary>The exact command to reproduce the issue</summary>

```bash
kopf run --standalone handlers.py
```
</details>

<details><summary>The full output of the command that failed</summary>

```
[2020-07-29 22:06:33,768] kopf.activities.star [INFO    ] Activity 'startup' succeeded.
[2020-07-29 22:06:33,768] kopf.reactor.activit [INFO    ] Initial authentication has been initiated.
[2020-07-29 22:06:33,769] kopf.activities.auth [INFO    ] Activity 'login_via_pykube' succeeded.
[2020-07-29 22:06:33,769] kopf.activities.auth [INFO    ] Activity 'login_via_client' succeeded.
[2020-07-29 22:06:33,770] kopf.reactor.activit [INFO    ] Initial authentication has finished.
[2020-07-29 22:06:51,218] kopf.objects         [ERROR   ] [default/test-cluster] Handler 'create' failed permanently: Missing required ConfigMap
[2020-07-29 22:06:51,223] kopf.objects         [INFO    ] [default/test-cluster] All handlers succeeded for creation.
```
</details>

## Environment

* Kopf version: 0.27
* Kubernetes version: 1.18
* Python version: 3.8.3
* OS/platform: Fedora 32

<details><summary>Python packages installed</summary>

```
```
</details>

**Comment by nolar** [comment]
Thanks for reporting. 

It seems like improper phrasing in the logs. The logic is as intended.

`kopf.PermanentError` (aliased as legacy `kopf.HandlerFatalError`) tells the framework to give up on this specific handler, and never retry it again. Once it is raised —assuming it is the only handler— there are no more handlers to try either now or later, including this same failed handler's retries, and so the handling cycle ends. 

This is marked by a phrase "all handlers succeeded" meaning that "all handlers have been tried and all of them have done what they could and I have nothing to do and bye-bye". This is the opposite to the case when there are handlers left to try/retry later until this handling cycle is finished.

A "handling cycle" is an internal term for full processing of a single cause (creation, update, deletion) with all matching handlers. I'm not sure if the "handling cycle" term should be exposed to the user.

I would be happy if you suggest a better phrasing that makes it more clear and less confusing.

**Comment by moltzaum** [comment]
Supposing the response is not conditional I think it is _slightly_ better phrasing to simply say: `All handlers finished for creation` or `All handlers completed for creation`. Or since not 'all' handlers may be run, just drop the 'all'.

But I do think it should have conditional phrasing. I think omission works since there is a previous message that says what failed. If an explicit message is still wanted though, maybe: `Handlers for creation aborted`

**Comment by nolar** [comment]
Solved https://github.com/nolar/kopf/pull/540, available since `kopf>=0.28rc2`
